### DISCUSSION TITLE
sha1_blockdata_order crash in Openssl 1.0.2u

### DISCUSSION THREAD

**Discussion by vinny20** [openssl/openssl#21994] [Q&A] [answered]
I am getting memory crash with the below callstack:

0x00002b8505a5d477 in sha1_block_data_order ()
0x00002b8505a5ea56 in SHA1_Update ()
0x00002b8505aaa396 in ssleay_rand_add ()
0x00002b8505aaa728 in ssleay_rand_bytes ()

Anyone face this same crash ??
Please let me know if anyone knows how to resolve this or any CVE for resolving this issue.

**Comment by paulidale** [comment, marked as answer]
My guess would be that something elsewhere is corrupting memory.
It could be OpenSSL, it could be your application.  Try running with valgrind or similar or building with address sanitisation enabled.

OpenSSL 1.0.2 is also out of support.

**Reply by vinny20** [reply]
This is with windows environment.
